analytics_data_api/v0/tests/views/test_courses.py

```diff
@@ -652,6 +652,90 @@ def test_get_404(self):
         self.assertEquals(response.status_code, 404)
 
 
+class CourseUsersProblemDataTests(DemoCourseMixin, TestCaseWithAuthentication):
+    def _get_data(self, user_id=None):
+        return self.authenticated_get('/api/v0/courses/{}/users/{}/problem_data/'.format(self.course_id, user_id))
+
+    def test_get_problem_data(self):
+        user_id = 111
+        G(
+            models.UserProblemWeeklyData,
+            id=1,
+            week_ending="2015-08-20",
```

Comment:
@bradenmacdonald When I change `week_ending` only for the mock input I get test failures. This seems to make sense because the [serializer](https://github.com/open-craft/edx-analytics-data-api/blob/88f1200298882089596393311eb854bf4df1862f/analytics_data_api/v0/serializers.py#L311) isn't set up to translate from a `DateField` to a string for the `week_ending` field.

For now I've changed both the mock input and the expected output to use `datetime` objects; the tests are passing. Let me know if you still have doubts, I'll adjust the code in that case.